Makefile

```diff
@@ -118,38 +118,27 @@ CROSS_BUILD_TARGETS := \
 	bin/podman.cross.linux.mips64 \
 	bin/podman.cross.linux.mips64le
 
-.PHONY: all
-all: binaries docs
-
 .PHONY: default
 default: help
 
-define PRINT_HELP_PYSCRIPT
-import re, sys
-
-print("Usage: make <target>")
-cmds = {}
-for line in sys.stdin:
-	match = re.match(r'^([a-zA-Z_-]+):.*?## (.*)$$', line)
-	if match:
-	  target, help = match.groups()
-	  cmds.update({target: help})
-for cmd in sorted(cmds):
-		print(" * '%s' - %s" % (cmd, cmds[cmd]))
-endef
-export PRINT_HELP_PYSCRIPT
+.PHONY: all
+all: binaries docs
 
 # Dereference variable $(1), return value if non-empty, otherwise raise an error.
 err_if_empty = $(if $(strip $($(1))),$(strip $($(1))),$(error Required variable $(1) value is undefined, whitespace, or empty))
 
+# Extract text following double-# for targets, as their description.
+_HLP_TGTS_RX = '^[[:print:]]+:.*?\#\# .*$$'
+_HLP_TGTS_CMD = grep -E $(_HLP_TGTS_RX) $(MAKEFILE_LIST)
+_HLP_TGTS_LEN = $(shell $(_HLP_TGTS_CMD) | cut -d : -f 1 | wc -L)
+_HLPFMT = "%-$(_HLP_TGTS_LEN)s %s\n"
```

Comment:
Could we move these _inside_ the target?  That would prevent them from being executed outside.

Reply:
It would seriously uglify the `help` target (because of the semicolon-backslash thing) but I agree: it's uncomfortable to have these run each time.